crates/subspace-networking/src/create.rs

```diff
@@ -336,54 +287,15 @@ async fn build_transport(
         MemoryTransport::default().or_transport(transport)
     };
 
-    if let Some(relay_transport) = relay_transport {
-        let transport = OrTransport::new(relay_transport, transport);
-        Ok(upgrade_transport(
-            transport,
-            &keypair,
-            timeout,
-            yamux_config,
-        ))
-    } else {
-        Ok(upgrade_transport(
-            transport,
-            &keypair,
-            timeout,
-            yamux_config,
-        ))
-    }
-}
-
-// Upgrades a provided transport with a multiplexer and a Noise authenticator.
-fn upgrade_transport<T, Output, Error, Listener, ListenerUpgrade, Dial>(
-    transport: T,
-    keypair: &identity::Keypair,
-    timeout: Duration,
-    yamux_config: YamuxConfig,
-) -> Boxed<(PeerId, StreamMuxerBox)>
-where
-    T: Transport<
-            Output = Output,
-            Error = Error,
-            Listener = Listener,
-            ListenerUpgrade = ListenerUpgrade,
-            Dial = Dial,
-        > + Send
-        + 'static,
-    Output: Unpin + Send + AsyncWrite + AsyncRead + 'static,
-    Error: Sync + Send + 'static,
-    Listener: Send + 'static,
-    ListenerUpgrade: Send + 'static,
-    Dial: Send + 'static,
-{
+    let transport = OrTransport::new(relay_transport, transport);
```

Comment:
Why do we add relay transport unconditionally? Isn't it redundant for relay servers? Probably, it's connected to your comment: `Various simplifications for cases that are not strictly necessary (for instance circuit relay server wasn't able to connect to peers that are themselves behind circuit relay 🤷)`

Reply:
It is connected to that comment, yes, you need a client to be able to connect to relay clients. Inability for relay servers to do this seemed like an arbitrary restriction that wasn't really necessary either.